src/overlays/actors/ovl_En_Sda/z_en_sda.c

```diff
@@ -21,17 +24,376 @@ const ActorInit En_Sda_InitVars = {
+                            temp_v1 = temp_t1 + phi_a3;
+                            if ((temp_v1 >= 0) && (temp_v1 < 0x1000)) {
+                                shadowTexture[temp_v0 + temp_v1] = 255;
+                            }
+                        }
+                    }
+                }
+                j++;
+            } while (j < 6);
+        }
+    }
+    if (BREG(61) == 1) {
+        for (shadowTextureTemp = shadowTexture, i = 0; i < 0x1000; i++, shadowTextureTemp++) {
+            if (*shadowTextureTemp != 0) {
+                *shadowTextureTemp = -((i >> 6) * (BREG(60) + 4)) + (255 - BREG(61));
+            }
+        }
+    }
+    osSyncPrintf("SDA CONT 4\n");
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sda/EnSda_Update.s")
+void func_80AF9C70(u8* shadowTexture, Player* player, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 tempx;
+    f32 tempz;
+    s16 phi_s1;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sda/EnSda_Draw.s")
+    OPEN_DISPS(gfxCtx, "../z_en_sda.c", 826);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sda/func_80AF8F60.s")
+    if (1) {}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sda/func_80AF95C4.s")
+    osSyncPrintf("SDA D 1\n");
+    func_80094044(globalCtx->state.gfxCtx);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x00, 0x00, 0, 0, 0, (BREG(52) + 50));
+    gDPSetEnvColor(POLY_XLU_DISP++, 0, 0, 0, 0);
+    Matrix_Translate(player->actor.posRot.pos.x, player->actor.groundY, player->actor.posRot.pos.z, MTXMODE_NEW);
+    Matrix_RotateY(BREG(51) / 100.0f, MTXMODE_APPLY);
+    Matrix_Scale(1.0f, 1.0f, (BREG(63) / 10.0f) + 1.0f, MTXMODE_APPLY);
+    tempx = (BREG(62) / 10.0f) + 2.0f;
+    tempz =
+        ((player->actor.posRot.pos.y - player->actor.groundY + BREG(54)) * (BREG(55) - 5) / 10.0f) + BREG(58) - 20.0f;
+    Matrix_Translate(tempx, 0.0f, tempz, MTXMODE_APPLY);
+    Matrix_Scale(((BREG(56) - 250) / 1000.0f) + 0.6f, 1.0f, ((BREG(59) - 250) / 1000.0f) + 0.6f, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_sda.c", 860), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, D_80AFA3D8);
+    gDPSetTextureImage(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 1, shadowTexture);
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 0, 0x0000, G_TX_LOADTILE, 0, G_TX_NOMIRROR | G_TX_CLAMP, 6,
+               G_TX_NOLOD, G_TX_NOMIRROR | G_TX_CLAMP, 6, G_TX_NOLOD);
+    gDPLoadSync(POLY_XLU_DISP++);
+    gDPLoadBlock(POLY_XLU_DISP++, G_TX_LOADTILE, 0, 0, 2047, 256);
+    gDPPipeSync(POLY_XLU_DISP++);
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_8b, 8, 0x0000, G_TX_RENDERTILE, 0, G_TX_NOMIRROR | G_TX_CLAMP, 6,
+               G_TX_NOLOD, G_TX_NOMIRROR | G_TX_CLAMP, 6, G_TX_NOLOD);
+    gDPSetTileSize(POLY_XLU_DISP++, G_TX_RENDERTILE, 0, 0, 0x00FC, 0x00FC);
```

Comment:
could be mistaken but i believe this can be shrunk down into a LoadTextureBlock macro

Reply:
So it can.